### DISCUSSION TITLE
How to do Inference profiling of the model in C++ (Android)?

### DISCUSSION THREAD

**Discussion by martinkorelic** [microsoft/onnxruntime#22742] [API Q&A]
I have an implemented model in C++ along with inference generation, however when I set to enable profiling on InferenceSession, do a single inference pass and then try to end profiling, I get no output json in the defined profile path. The returned profile path string of SessionEndProfiling "\0". I have tried to use also EndProfilingAllocated, tried exiting the process or just releasing the InferenceSession, however no file was created or saved.

It should be noted that I am doing this on Android and using EnableProfiling with InferenceSession, which is created successfully as it returns a null pointer.

There is a similar issue to [this one](https://github.com/microsoft/onnxruntime/issues/10975), however it was not answered.
If anyone has any successful experience of profiling on C++ using onnxruntime on mobile feel free to guide me in the right direction.

**Comment by skottmckay** [comment]
Setting SessionOptions `enable_profiling` to the profile filename prefix has worked for me on Android recently.

It was enabled via a genai config setting, which results in this call:

https://github.com/microsoft/onnxruntime-genai/blob/c5745fd6d91a2ba2d94c6a52f6255652dde02508/src/models/model.cpp#L359

Does profiling work with other models? e.g. something from https://onnx.ai/models/
